src/celeritas/user/detail/ActionDiagnosticLauncher.hh

```diff
@@ -0,0 +1,71 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2023 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/user/detail/ActionDiagnosticLauncher.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "corecel/Assert.hh"
+#include "corecel/Macros.hh"
+#include "corecel/math/Atomics.hh"
+#include "celeritas/global/CoreTrackData.hh"
+#include "celeritas/global/CoreTrackView.hh"
+
+namespace celeritas
+{
+namespace detail
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Kernel launcher for tallying post-step actions by particle type.
+ */
+struct ActionDiagnosticLauncher
```

Comment:
With #737 we should be able to reuse the templated launcher but that can be a follow-up PR which does it for multiple files

Reply:
Using the templated launcher now.